src/promptflow/promptflow/_sdk/entities/_eager_flow.py

```diff
@@ -59,3 +59,37 @@ def _dump_for_validation(self) -> Dict:
         return self._data
 
     # endregion SchemaValidatableMixin
+
+    @classmethod
+    def _resolve_entry_file(cls, entry: str, working_dir: Path) -> Optional[str]:
+        """Resolve entry file from entry.
+        If entry is a local file, e.g. my.local.file:entry_function, return the local file: my/local/file.py
+            and executor will import it from local file.
+        Else, assume the entry is from a package e.g. external.module:entry, return None
+            and executor will try import it from package.
+        """
+        try:
+            entry_file = f'{entry.split(":")[0].replace(".", "/")}.py'
+        except Exception as e:
+            raise UserErrorException(f"Entry function {entry} is not valid: {e}")
+        entry_file = working_dir / entry_file
+        if entry_file.exists():
+            return entry_file.resolve().absolute().as_posix()
+        # when entry file not found in working directory, return None since it can come from package
+        return None
+
+    def _init_executable(self, **kwargs):
```

Comment:
Remove core flow's _init_executable?

Reply:
Let me do this in next PR

Reply:
https://github.com/microsoft/promptflow/pull/2265